apps/transport/lib/transport_web/live/discussions_live.ex

```diff
@@ -89,8 +89,14 @@ defmodule TransportWeb.DiscussionsLive do
   def discussion_should_be_closed?(%{"closed" => closed}) when not is_nil(closed), do: true
 
   def discussion_should_be_closed?(%{"discussion" => comment_list}) do
-    {:ok, latest_comment_datetime, 0} = List.first(comment_list)["posted_on"] |> DateTime.from_iso8601()
-    DateTime.utc_now() |> Timex.shift(months: -2) |> DateTime.compare(latest_comment_datetime) == :gt
+    latest_comment_datetime =
+      comment_list
+      |> Enum.map(&DateTime.from_iso8601(&1["posted_on"]))
+      |> Enum.map(&elem(&1, 1))
+      |> Enum.max(DateTime)
```

Comment:
```suggestion
    latest_comment_datetime =
      Enum.max_by(comment_list, fn %{"posted_on" => posted_on} ->
         {:ok, datetime, 0} = DateTime.from_iso8601(posted_on)
         datetime
      end, DateTime)
```

me parait légèrement mieux, évite de parcourir 3 fois l'array (2 maps + le max) et le pattern matching mettrant en évidence une erreur si elle survient (alors que `elem(&1, 1)) va passer sous silence une potentielle erreur.

PS : code non testé !

Reply:
https://hexdocs.pm/elixir/Enum.html#max_by/4

https://hexdocs.pm/elixir/DateTime.html

> Remember, comparisons in Elixir using [==/2](https://hexdocs.pm/elixir/Kernel.html#==/2), [>/2](https://hexdocs.pm/elixir/Kernel.html#%3E/2), [</2](https://hexdocs.pm/elixir/Kernel.html#%3C/2) and friends are structural and based on the DateTime struct fields. For proper comparison between datetimes, use the [compare/2](https://hexdocs.pm/elixir/DateTime.html#compare/2) function. The existence of the [compare/2](https://hexdocs.pm/elixir/DateTime.html#compare/2) function in this module also allows using [Enum.min/2](https://hexdocs.pm/elixir/Enum.html#min/2) and [Enum.max/2](https://hexdocs.pm/elixir/Enum.html#max/2) functions to get the minimum and maximum datetime of an [Enum](https://hexdocs.pm/elixir/Enum.html).

Reply:
En fait j’ai utilisé `max` au lieu de `max_by`, parce que `max_by` te renvoie l’élément de l’enumerable initial, donc un commentaire (complet) tel qu’il est dans le retour de l’API. Et donc il faudra une deuxième fois aller y récupérer la string de la date et la parser avant de la comparer à two_months_ago. Du coup j’ai préféré transformer une seule fois en DateTime. À moins qu’il y ait quelque chose que je rate ?